common/functions/src/scalars/expressions/expression.rs

```diff
@@ -32,11 +32,17 @@ impl ToCastFunction {
         // TODO(zhyass): complete DateTime, e.g. toDateTime64(1640019661000, 3, 'UTC').
         features = match type_name {
             "Boolean" => features.num_arguments(1).bool_function(),
-            // "DateTime" | "DateTime32" => features.variadic_arguments(1, 2),
-            "DateTime" => features.variadic_arguments(1, 3),
+            "TimeStamp" | "DateTime" => features.variadic_arguments(1, 3),
             _ => features.num_arguments(1),
         };
 
+        // special case: datetime is alias to timestamp, change cast type to timestamp
+        let type_name = if type_name == "DateTime" {
```

Comment:
we can set `DateTime` as an alias of `Timestamp` instead of this special case.

Reply:
agree

Reply:
reasonable!

Reply:
resolved.